src/App.jsx

```diff
@@ -16,11 +16,11 @@ function App() {
       .catch((error) => {
         throw new Error(`Error: ${error}`)
       })
-  }, [])
+  }, [setUser])
```

Comment:
Super close here! This is still making the fetch call from `App.js`, rather than from within the `UserContext`. The `useEffect` should be moved over into the `UserContext` custom hook as well.